db/revision.go

```diff
@@ -189,18 +189,62 @@ func (db *DatabaseContext) getOldRevisionJSON(docid string, revid string) ([]byt
 	return data, err
 }
 
-func (db *Database) setOldRevisionJSON(docid string, revid string, body []byte) error {
-	base.Debugf(base.KeyCRUD, "Saving old revision %q / %q (%d bytes)", base.UD(docid), revid, len(body))
+// Makes a backup of revision body for use by delta sync, and in-flight replications requesting an old revision.
+// Backup policy depends on whether delta sync and/or shared_bucket_access is enabled
+//   delta=false
+//      - old revision stored, with expiry OldRevExpirySeconds
+//   delta=true, shared_bucket_access=true
+//      - new revision stored (as duplicate), with expiry rev_max_age_seconds
+//   delta=true, shared_bucket_access=false
+//      - old revision stored, with expiry rev_max_age_seconds
+func (db *Database) backupRevisionJSON(docId, newRevId, oldRevId string, newBody Body, oldBody []byte) {
+
+	if !db.DeltaSyncEnabled() {
+		db.setOldRevisionJSON(docId, oldRevId, oldBody, db.Options.OldRevExpirySeconds)
+		return
+	}
+
+	if db.UseXattrs() {
+		newBodyBytes, marshalErr := json.Marshal(newBody)
+		if marshalErr != nil {
+			base.Warnf(base.KeyAll, "Unable to marshal new revision body during backupRevisionJSON: doc=%q rev=%q err=%v ", base.UD(docId), newRevId, marshalErr)
+			return
+		}
+		db.setOldRevisionJSON(docId, newRevId, newBodyBytes, db.Options.DeltaSyncOptions.RevMaxAgeSeconds)
+
+		// Refresh the expiry on the previous revision backup
+		db.refreshPreviousRevisionBackup(docId, oldRevId, oldBody, db.Options.DeltaSyncOptions.RevMaxAgeSeconds)
+	} else {
+		db.setOldRevisionJSON(docId, oldRevId, oldBody, db.Options.DeltaSyncOptions.RevMaxAgeSeconds)
```

Comment:
assign err to `_` for intentional ignore

Reply:
Agree - fixed.